Update view after episode deletion when downloaded filter is enabled

Niffler opened the pull request:
Fixes #4548

ByteHamster commented:
Thanks for working on this. In other places, AntennaPod uses `events` to react to changes in the database. An advantage of that is that items are only removed from view when they are actually removed (maybe there are cases where removing fails). Also, it looks like your code always removes the item, even if the list is not filtered. 

I suggest to make use of the function `shouldUpdatedItemRemainInList` that can be overridden in `AllEpisodesFragment`. That method can then check if the filter is enabled and then return `false` if the item no longer has media.

Niffler commented:
Thanks for reviewing my pull request. I reverted the previous changes and implemented the view update according to your suggestions, using the `shouldUpdatedItemRemainInList` method. Thereby episodes are not removed from the view when the downloaded filter is not enabled.

ByteHamster commented on app/src/main/java/de/danoeh/antennapod/fragment/AllEpisodesFragment.java:
```diff
@@ -89,6 +89,18 @@ protected void updateFilter(Set<String> filterValues) {
+        SharedPreferences prefs = getActivity().getSharedPreferences(PREF_NAME, Context.MODE_PRIVATE);
+        FeedItemFilter feedItemFilter = new FeedItemFilter(prefs.getString(PREF_FILTER, ""));
+
+        if (feedItemFilter.isShowDownloaded()) {
```
Items can be updated for other reasons, too. This line is missing a check if the item is actually no longer downloaded.

ByteHamster commented:
Thanks a lot! The fix will be shipped in the 2.1.x versions because I don't think the bug is critical enough for a new update.
